src/ServiceControl.Transports.PostgreSql/PostgreSqlTable.cs

```diff
@@ -7,13 +7,14 @@ public PostgreSqlTable(string name, string schema)
     {
         //HINT: The query approximates queue length value based on max and min of the table sequence.
         fullTableName = $"\"{schema}\".\"{name}\"";
-        //NOTE: Postgres doesn't have NOLOCK since it utilises snapshot isolation by default
+        //      As a result, we want to skip rows locked by other transactions with SKIP LOCKED query hint.
         LengthQuery = $$"""
                         SELECT CASE WHEN (EXISTS (SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = '{{schema}}' AND TABLE_NAME = '{{name}}')) THEN
-                          COALESCE(cast(max(seq) - min(seq) + 1 AS int), 0)
+                          COALESCE(cast((SELECT seq FROM {{schema}}."{{name}}" ORDER BY seq DESC LIMIT 1 FOR UPDATE SKIP LOCKED)
+                                      - (SELECT seq FROM {{schema}}."{{name}}" ORDER BY seq ASC  LIMIT 1 FOR UPDATE SKIP LOCKED) + 1 AS int), 0)
                         ELSE
                           -1
```

Comment:
this doesn't need to have the column name `Id` anymore?

Reply:
Good catch, I did not notice that! 

Looking at the code it seems that we don't care about the column name, only about the index:

https://github.com/Particular/ServiceControl/blob/5ccfb932d7c03c19a4e87193cb18ec3631091f08/src/ServiceControl.Transports.PostgreSql/QueueLengthProvider.cs#L112

Smoke tests that I've performed also indicate that this works fine.